src/sleep/sleep.c

```diff
@@ -169,12 +169,12 @@ static int lock_all_homes(void) {
         return log_debug("Successfully requested locking of all home directories.");
 }
 
-static int execute(char **modes, char **states) {
+static int execute(char **modes, char **states, const char *action) {
         char *arguments[] = {
                 NULL,
                 (char*) "pre",
                 arg_verb,
-                NULL
+                (char *) action
```

Comment:
The final NULL is required and will make nasty things happen if missing. This array is passed to ```execv``` which states:

> The array of pointers must be terminated by a null pointer.